tasks/add-unique-granuleID/README.md

```diff
@@ -0,0 +1,106 @@
+# @cumulus/add-unique-granuleID
+
+This is a [Cumulus](https://nasa.github.io/cumulus) task which takes the following actions on each granule in an incoming set of payload granules:
+
+- Adds the existing `granuleId` to the `producerGranuleId` key
+- Updates the existing `granuleId` field to a 'unique' value based on the algorithim used in @cumulus/ingest `generateUniqueGranuleId`
+
+## Usage
+
+This lambda takes the following input and config objects, derived from workflow configuration using the [Cumulus Message Adapter](https://github.com/nasa/cumulus-message-adapter/blob/master/CONTRACT.md) to drive configuration from the full cumulus message:
+
+### Input
+
+The input takes a list of 'pre-sync' or Cumulus API formatted granules, with the following required fields:
+
+```json
+{
+  "granules": [
+    {
+      "granuleId": "foobar",
+      "collectionId": "someCollection___001"
```

Comment:
I think we have an open question about what `producerGranuleId` is at this point. Would the field be present but undefined here? 

We may want to re-visit these docs after we decide what we're doing with that field. If it is there, it'd be nice to call it out in the incoming payload docs so it's clearer what happens at the output.

Reply:
I guess the clarity I'm searching for is when `producerGranuleId` is actually added... which we haven't totally decided.

Reply:
Is there ambiguity here with respect to design?  We're adding producerGranuleId to the message payload/granule object output.  That's not the same thing as what the framework does with the object - but this task component is doing exactly what it's intended to do, right?   That's more a function of the architecture - this document is supposed to explain what this task does, absent composition concerns.

Reply:
The header for this section says "required" fields so it's probably fine.